File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/OnScreenGamePad.java
package com.nopalsoft.sokoban.scene2d;

import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.nopalsoft.sokoban.Assets;
import com.nopalsoft.sokoban.game.GameScreen;

public class OnScreenGamePad extends Table {

    GameScreen gameScreen;

    Button btUp, btDown, btLeft, btRight;

    public OnScreenGamePad(GameScreen oScreen) {
        gameScreen = oScreen;

        getColor().a = .4f;


        btUp = new Button(Assets.btUp, Assets.btUpPress);
        btUp.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.up();
            }
        });

        btDown = new Button(Assets.btDown, Assets.btDownPress);
        btDown.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.down();
            }
        });

        btLeft = new Button(Assets.buttonIzq, Assets.btIzqPress);
        btLeft.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.left();
            }
        });

        btRight = new Button(Assets.buttonDer, Assets.buttonDerPress);
        btRight.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                gameScreen.right();
            }
        });


        int buttonSize = 75;
        defaults().size(buttonSize);

        add(btUp).colspan(2).center();
        row();
        add(btLeft).left();
        add(btRight).right().padLeft(buttonSize / 1.15f);
        row();
        add(btDown).colspan(2).center();
        pack();
    }


}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/OnScreenGamePad.kt
package com.nopalsoft.sokoban.scene2d

import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.Button
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.nopalsoft.sokoban.Assets
import com.nopalsoft.sokoban.game.GameScreen

/**
 * The game-pad on screen of the game.  It allows mobile phone users to control the player.
 */
class OnScreenGamePad(val screen: GameScreen) : Table() {

    private val buttonUp = Button(Assets.btUp, Assets.btUpPress)
    private val buttonDown = Button(Assets.btDown, Assets.btDownPress)
    private val buttonLeft = Button(Assets.buttonIzq, Assets.btIzqPress)
    private val buttonRight = Button(Assets.buttonDer, Assets.buttonDerPress)

    init {
        color.a = 0.4F
        buttonUp.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                screen.up()
            }
        })
        buttonDown.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                screen.down()
            }
        })
        buttonLeft.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                screen.left()
            }
        })
        buttonRight.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                screen.right()
            }
        })

        val buttonSize = 75F
        defaults().size(buttonSize)

        add(buttonUp).colspan(2).center()
        row()
        add(buttonLeft).left()
        add(buttonRight).right().padLeft(buttonSize / 1.15F)
        row()
        add(buttonDown).colspan(2).center()
        pack()
    }
}